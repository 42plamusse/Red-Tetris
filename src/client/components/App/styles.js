import styled from "styled-components";

export const StyledContainer = styled.div`
  position: fixed;
  width: 100%;
  display: flex;
  flex-direction: column;
`;
