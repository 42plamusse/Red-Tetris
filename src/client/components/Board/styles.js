import styled, { css } from "styled-components";
import { PURE_ORANGE, CRIMSON } from "../../../constants/colors";

export const StyledAspectRatio = styled.div`
  padding-top: 200%;
  width: 100%;
  height: 0;
  position: relative;
  // box-sizing: border-box;
`;

export const StyledAspectRatioContainer = styled.div`
  flex-basis: 25%;
  // box-sizing: border-box;
  text-align: center;
  max-width: 500px;
  min-width: 325px;
  ${props =>
    props.isSpectre &&
    css`
      flex-basis: 30%;
      padding: 1em;
      min-width: 100px;
      max-width: 150px;
    `};
`;

export const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 0px;
  border: 4px solid black;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  // box-sizing: border-box;
  ${props =>
    props.isSpectre &&
    css`
      border: 2px solid black;
      background: linear-gradient(${PURE_ORANGE} 0%, ${CRIMSON} 100%);
    `};
`;

export const StyledRow = styled.div`
  flex: 1;
  display: flex;
`;
