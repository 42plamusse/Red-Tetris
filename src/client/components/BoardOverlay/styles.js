import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  justify-content: center;
  border: 3px solid white;
  border-radius: 5px;
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: -3px;
  height: 86vh;
  width: 43vh;
`;

export const StyledGameOverText = styled.h1`
  align-self: center;
  font-size: 5vh;
`;
