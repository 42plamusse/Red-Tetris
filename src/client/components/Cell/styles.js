import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  flex: 1 10%;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  border: 1px solid black;
  ${props =>
    props.color &&
    css`
      box-sizing: border-box;
      -moz-box-sizing: border-box;
      -webkit-box-sizing: border-box;
      border: 5px outset ${props => props.color};
      background-color: ${props => props.color};
    `};
`;
