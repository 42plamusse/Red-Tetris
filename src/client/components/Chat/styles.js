import styled from "styled-components";

export const StyledMainContainer = styled.div`
  flex: 4;
  padding: 20px;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.9);
  // border: 5px solid white;
  border-radius: 2px;
`;
