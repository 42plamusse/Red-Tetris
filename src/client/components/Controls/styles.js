import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  div {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    kbd {
      font-size: 2vh;
      border-radius: 3px;
      padding: 1px 2px 0;
      border: 3px solid white;
    }
    span {
      font-size: 1.5vh;
    }
  }
  h2 {
    margin-top: 40px;
    align-self: center;
  }
`;
