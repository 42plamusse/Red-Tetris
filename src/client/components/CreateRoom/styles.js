const styles = {
  createRoomContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  }
};

export default styles;
