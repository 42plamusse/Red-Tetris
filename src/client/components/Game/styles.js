import styled from "styled-components";

export const StyledGameContainer = styled.div`
  display: flex;
  margin: 20px;
`;

export const StyledBoardsContainer = styled.div`
  box-sizing: border-box;
  display: flex;
  width: 100%;
  height: 100%;
`;
