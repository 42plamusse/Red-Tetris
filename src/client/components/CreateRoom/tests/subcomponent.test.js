import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CreateRoomSub from "../subcomponent";
import { StyledForm, StyledTextField } from "../styles";

const props = {
  playerName: "Bob"
};

describe("<CreateRoomSub />", () => {
  test("should render without crashing", () => {
    const wrapper = shallow(<CreateRoomSub {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should update roomName and set error to true with a too long playerName", () => {
    const wrapper = shallow(<CreateRoomSub {...props} />);
    const roomName = "RoomNameTooLong";
    const event = { target: { value: roomName } };

    wrapper.find(StyledTextField).simulate("change", event);
    expect(wrapper.find(StyledTextField).props().value).toBe(roomName);
    expect(wrapper.find(StyledTextField).props().error).toBeTruthy();
  });

  test("should update roomName and set error to true with an unvalid character", () => {
    const wrapper = shallow(<CreateRoomSub {...props} />);
    const roomName = "Room$";
    const event = { target: { value: roomName } };

    wrapper.find(StyledTextField).simulate("change", event);
    expect(wrapper.find(StyledTextField).props().value).toBe(roomName);
    expect(wrapper.find(StyledTextField).props().error).toBeTruthy();
  });

  test("should update playerName, set error to false and call connectPlayer on Submit with a valid input", () => {
    const wrapper = shallow(<CreateRoomSub {...props} />);
    const roomName = "ValidRoom";
    const event = { target: { value: roomName } };

    wrapper.find(StyledTextField).simulate("change", event);
    expect(wrapper.find(StyledTextField).props().value).toBe(roomName);
    expect(wrapper.find(StyledTextField).props().error).not.toBeTruthy();
    wrapper.find(StyledForm).simulate("submit", {
      preventDefault: () => {}
    });
    expect(window.location.hash).toBe(
      "#" + roomName + "[" + props.playerName + "]"
    );
  });
});
