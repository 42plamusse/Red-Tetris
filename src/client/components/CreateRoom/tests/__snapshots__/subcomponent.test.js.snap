// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoomSub /> should render without crashing 1`] = `
<styled.form
  id="form"
  onSubmit={[Function]}
>
  <styled.div
    id="register"
  >
    <WithStyles(ForwardRef)
      autoComplete="off"
      error={null}
      helperText=""
      id="name"
      label="Insert room name"
      onChange={[Function]}
      spellCheck="false"
      value=""
    />
    <styled.button
      disabled={true}
      id="enter"
    >
      <p>
         Press Enter 
      </p>
    </styled.button>
  </styled.div>
</styled.form>
`;
